types/hooks.d.ts

```diff
@@ -65,6 +86,12 @@ export type UseSpringProps<DS extends UseSpringBaseProps> = Merge<
 >
 
 // there's a third value in the tuple but it's not public API (?)
+export function useSpring<DS extends CSSProperties>(
+  values: UseSpringProps<DS & CSSProperties>
```

Comment:
This trick lets CSSProperties show up on autocomplete by default (as long as no property that you add is incompatible with CSSProperties itself; then it falls back to the other overloads)